packages/syft/src/syft/service/network/network_service.py

```diff
@@ -124,36 +130,38 @@ def create_or_update_peer(
             Result[NodePeer, str]: The updated or added peer if the operation
             was successful, or an error message if the operation failed.
         """
-        valid = self.check_type(peer, NodePeer)
-        if valid.is_err():
-            return SyftError(message=valid.err())
-        existing: Result | NodePeer = self.get_by_uid(
-            credentials=credentials, uid=peer.id
+        self.check_type(peer, NodePeer).unwrap()
+        try:
```

Comment:
can we rewrite this logic with just .is_err(), I think this is slightly harder to read because it zips the code of the if and the else case